pricegraph/src/orderbook.rs

```diff
@@ -173,14 +179,23 @@ impl Orderbook {
                 if let Some(base_index) = path.iter().position(|node| *node == base) {
                     let (bid, ask) = (&path[0..base_index + 1], &path[base_index..]);
 
-                    overlap.asks.push(
-                        self.find_transitive_order(ask)
-                            .expect("ask transitive path not found after being detected"),
-                    );
-                    overlap.bids.push(
-                        self.find_transitive_order(bid)
-                            .expect("bid transitive path not found after being detected"),
-                    );
+                    let (capacity, transitive_price) = self
+                        .fill_path(ask)
```

Comment:
I really don't see the difference between filling the two halfs of the path individually here and `continue` vs. filling the full cycle in one go outside of this if statement (as it was before). I read all your comments but I don't get how e.g. the new unit test was not passing before.

Reply:
Ok, I think I see it now (after reading the associated issue). We are filling the "entirety" of the order and not just the potential fraction that is part of the ring.